src/cloud-cli/meltano/cloud/cli/history.py

```diff
@@ -0,0 +1,139 @@
+"""Meltano project execution history."""
+
+from __future__ import annotations
+
+import asyncio
+import datetime
+import json
+import typing as t
+
+import click
+import tabulate
+
+from meltano.cloud.api.client import MeltanoCloudClient
+from meltano.cloud.cli.base import cloud
+
+if t.TYPE_CHECKING:
+    from meltano.cloud.api.config import MeltanoCloudConfig
+    from meltano.cloud.api.types import CloudExecution
+
+PAGE_SIZE = 100
+
+
+async def _get_history(
+    config: MeltanoCloudConfig,
+    schedule_filter: str | None,
+    limit: int,
+) -> list[CloudExecution]:
+    """Get a Meltano project execution history in Meltano Cloud.
+
+    Args:
+        config: The meltano config to use
+        schedule_filter: Used to filter the history by schedule name.
+        limit: The maximum number of history items to return.
+
+    Returns:
+        The execution history.
+    """
+    page_token = None
+    page_size = min(limit, PAGE_SIZE)
+    results: list[CloudExecution] = []
+
+    async with MeltanoCloudClient(config=config) as client:
+        while True:
+            response = await client.get_execution_history(
+                schedule=schedule_filter,
+                page_size=page_size,
+                page_token=page_token,
+            )
+
+            results.extend(response["results"])
+
+            if response["pagination"] and len(results) < limit:
+                page_token = response["pagination"]["next_page_token"]
+            else:
+                break
+
+    return results[:limit]
+
+
+def _format_history_table(history: list[CloudExecution]) -> str:
+    """Format the history as a table.
+
+    Args:
+        history: The history to format.
+
+    Returns:
+        The formatted history.
+    """
+    table = []
+    headers = ["Execution ID", "Schedule Name", "Executed At", "Result", "Duration"]
+
+    for execution in history:
+        start_time = datetime.datetime.fromisoformat(execution["start_time"])
+
+        if execution["end_time"]:
+            end_time = datetime.datetime.fromisoformat(execution["end_time"])
+            td = end_time - start_time
+            duration = f"{td.total_seconds() / 60:.2f} minutes"
```

Comment:
I think I'd prefer seeing the runtime as a whole number of seconds with no unit suffix, and then name the column `Duration (s)` or `Duration (seconds)`. Opinions? @aaronsteers @tayloramurphy

Reply:
@WillDaSilva I have a preference for something like 6m42s - but we can't really assume runs will all be that short. 

Can we just do `00:06:42` as the format? Seems like that covers most cases.

Reply:
@taylormurphy yeah, we can do that